### PR TITLE
Add Metrics: track latest height + broadcast errors

### PR DESCRIPTION
**Pull request by boojamya**
This PR adds `cctp_relayer_chain_latest_height` and `cctp_relayer_broadcast_errors_total` Prometheus metrics. 

See README for details on each metric.

Unrelated to metrics,
This PR has some minor readme updates and fixes a few spelling mistakes. 
Apologies if that makes the review more confusing!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,9 +30,17 @@ After that, it will flush from the last stored height - lookback period up until
...
-| cctp_relayer_wallet_balance | Current balance of a relayer wallet in Wei.<br><br>Noble balances are not currently exported b/c `MsgReceiveMessage` is free to submit on Noble.   | Gauge    |
+| **Exported Metric**                 | **Description**                                                                                                                                    | **Type** |
+|-------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------|----------|
+| cctp_relayer_wallet_balance         | Current balance of a relayer wallet in Wei.<br><br>Noble balances are not currently exported b/c `MsgReceiveMessage` is free to submit on Noble.   | Gauge    |
+| cctp_relayer_chain_latest_height    | Current height of the chain.                                                                                                                       | Gauge    |
+| cctp_relayer_broadcast_errors_total | The total number of failed broadcasts. Note: this is AFTER is retires `broadcast-retries` (config setting) number of times.                        | Counter  |
```
Comment (reecepbcups) [inline README.md]:
```suggestion
| cctp_relayer_broadcast_errors_total | The total number of failed broadcasts. Note: this is AFTER it retries `broadcast-retries` (config setting) number of times.                        | Counter  |
```

File: cmd/process_test.go [hunk 2]
```diff
@@ -24,7 +24,7 @@ func TestProcessNewLog(t *testing.T) {
 	sequenceMap := types.NewSequenceMap()
 	processingQueue = make(chan *types.TxState, 10)
 
-	go cmd.StartProcessor(context.TODO(), a, registeredDomains, processingQueue, sequenceMap)
+	go cmd.StartProcessor(context.TODO(), a, registeredDomains, processingQueue, sequenceMap, nil)
```
Comment (reecepbcups) [inline cmd/process_test.go:27]: every instance of `StartProcessor` in this PR now passes nil. Is the arg: sequenceMap still required for noble or ETH?

**Review by joelsmith-2019 — APPROVED** [review]
Nice work! Tested the metrics locally and they are displaying under the /metrics endpoint.

File: relayer/metrics.go [hunk 3]
```diff
@@ -10,25 +10,40 @@ import (
 )
 
 type PromMetrics struct {
-	WalletBalance *prometheus.GaugeVec
+	WalletBalance   *prometheus.GaugeVec
+	LatestHeight    *prometheus.GaugeVec
+	BroadcastErrors *prometheus.CounterVec
 }
 
+// cctp_relayer_broadcast_errors_total
```
Comment (reecepbcups) [inline relayer/metrics.go]: nit: comment

**Review by reecepbcups — APPROVED** [review]

File: cmd/process_test.go (hunk 2 above)
Comment (reecepbcups) [reply cmd/process_test.go:27]: yes, is used in Start(a *AppState) only looks like

Comment (boojamya) [reply cmd/process_test.go:27]: Yes. It was either passing nil or mocking out a no-op prometheus metrics for testing.

**Pull request merged**